### DISCUSSION TITLE
Privacy: harvesting information from text we've translate

### DISCUSSION THREAD

**Discussion by bintaniq** [nidhaloff/deep-translator#147] [Q&A]
Hi Nidhaloff,

Greetings, it is an honor to have a public discussion with you. 

If you don't mind, I would like to confirm one thing regarding privacy. Since the deep-translators is build on top of publicly available **free** version of translations library (Google, Mymemory Deepl, Qcri, Linguee, PONS, Yandex, Microsoft, Papago, Libre translators), can I confirm those library are allowed to harvest information from text we've translate?

If you might suggest, which translators that have highest/ distinct privacy policy among deep-translator library?

In the case of none of the library can ensure robust data privacy policy, I am afraid I might need to looks for paid options (i.e. cloud google translate, DeepL Pro, etc.)

Appreciate your feedback on this. Thanks :)

p.s. some articles related to free and paid translators:
[Article](https://proprivacy.com/blog/google-translate-privacy) about Google Translate privacy for free and paid version
Paid google translate policy [article](https://cloud.google.com/translate/data-usage)

**Comment by nidhaloff** [comment]
Hi @bintaniq If you mean whether deep-translator stores your data in any form, then the answer is no. Beyond that, it is really hard to know what the third parties are doing with the text. I can only answer this question regarding what deep-translator is doing, which I already answered. 

I assume using proxies would help, but not with your data since your data has to land in the translator server anyway.
